tests/js/multiparser-markdown/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -109,7 +109,7 @@ markdown\`
 
 =====================================output=====================================
 markdown\`
-  const cssString = css\\\`background-color: \\$\\{color('base')\\}\\\`;
+  const cssString = css\\\` background-color: \\$\\{color('base')\\}\\\`;
```

Comment:
This is the same change as https://github.com/prettier/prettier/pull/8140#discussion_r421791594 but in this case I think the change is not desirable. Fixing this would probably mean also making the the instance in the other comment go back to the previous formatting. I'll do that as it seems like the safest option for a dependency update.

Reply:
This is a really strange test though. Something that looks like JS is marked as Markdown. I need to investigate where this test came from.

Reply:
This test was added to check that backticks are correctly escaped in JS template literals formatted as Markdown. So I think this change is okay. Is there a reason to remove this whitespace?

Reply:
Ah okay, I wondered why it was such a strange string to test for. I don't have any objection to this change personally. It wouldn't make any semantic change to the embedded CSS anyway... I'm fine to leave this be if you are!

Reply:
No, turns out it's not okay to add this space. It [changes the output](https://spec.commonmark.org/dingus/?text=const%20cssString%20%3D%20css%60%0A%20%20%20%20background-color%3A%20%24%7Bcolor(%27base%27)%7D%0A%20%20%60%3B%0A%0Aconst%20cssString%20%3D%20css%60background-color%3A%20%24%7Bcolor(%27base%27)%7D%60%3B%0A%0Aconst%20cssString%20%3D%20css%60%20background-color%3A%20%24%7Bcolor(%27base%27)%7D%60%3B).

Reply:
I'm revisiting this one today. I'm really not sure what to do about it because prettier isn't actually adding the extra space here. It's not clear to me if `remark` was outputting a different AST in the past but currently the value of the AST's `inlineCode` includes the leading space. I'm wracking my brain for a way of dealing with this programatically but just cannot come up with anything.

Do you have any ideas @thorn0 or @fisker? The relevant code is here: https://github.com/prettier/prettier/blob/master/src/language-markdown/printer-markdown.js#L143:L143

On that line, for this particular example `node.value` includes the leading space and `gap === ""`

It seems like this _might_ be a difference between `remark` and `commonmark`. Comparing the AST output of the following:

```
const cssString = css`
  background-color: ${color('base')}
`;
```

Commonmark: https://spec.commonmark.org/dingus/?text=const%20cssString%20%3D%20css%60%0A%20%20background-color%3A%20%24%7Bcolor(%27base%27)%7D%0A%60%3B

vs

Remark: https://astexplorer.net/#/gist/0c3e1abd39c043261e6ec27a2ea2dd3a/df7fa0bfadaa16a75413b026547c700963470d01

You can see that the `inlineCode`'s `value` output by remark actually includes 2 leading spaces (not sure why this is being collapsed to a single space) while commonmark's `code` element has the spaces stripped.

This does indeed seem to be a change in `remark` as the current version of `prettier` using v5 does not include the leading spaces: [prettier master playground example (select the debug option to show the AST)](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEY1oMowBOAllAOaYC82uABgDpSaYBGAhmANZmEQCuUACYBaSABsIhJJgAkwcZIAUAcnZo4ygJQBfRrQDcIADQgIABxjF0yUG0K8A7gAU7CNMhBsxDtgE93JiyEHJxwMHhmHKRkyER8cCYAFjAAtmIA6onE8GiRYHB4btnEAG7Zvh44ASCk6oQwTsFkKWzIAGZe6iYAVmgAHgBCwVxheGwpcAAypHDtnQkgvX140WJwAIp8EPBzYl0gkYR1Hi2EnIIQDlDGBySw6cSCMInIABwADCZmvOrpwWYeb5wOolWYmACOW3gjXM7hQbDQwigcDgglRN0IcEhxExjTYzVaSA6ewW6hSxFihHiJjQqw2UNmRPmJhgbBYDyeLyQACYWcFiGJogBhCApFoeYEAVhufHUABU2XDifsSvEAJJCBDhMAkCwAQSEBF8a126m02iAA).

[When you do the same in the playground for this branch](https://deploy-preview-8140--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuc0DOMAEY1oMowBOAllAOaYC82uABgDpSaYBGAhmANZmEQCuUACYBaSABsIhJJgAkwcZIAUAcnZo4ygJQBfRrQDcjEABoQEAA4xi6ZKDaFeAdwAK9hGmQg2Yx2wCeHqYshByccDB45hykZMhEfHCmcAC2LHCCgukAMmzkfGxkcABikslsMFbknmx8MBAmIAAWMMliAOqNxPBoUWBweO5dxABuXX6eOIEgpOqEMM4hZGXIAGbe6qYAVmgAHgBCIVzheGzJcFmkcKvriSDbO3gxYnAAinwQ8NdiGyBRhLOeMqETiCCCOKANcwkWBtYiCGCNZAAZgALKYoRB1G0QuZPFC4LNhldTABHd7wBYWDwoNhoYRQODpdINQhwMnEVkLArLJBrb63dTJYhxQgJUxoJ6vclXXk3UwwNgsWHwxFIABM8pCxDEMQAwhBkjyQASAKwNPjqAAqiupfJ+wwSAEkhAgImASJYAIJCAh+Z5fdTabRAA), you can also see that the AST itself is different and matches the exact AST differences between `remark` in AST Explorer and commonmark.

Anyway, all of that to say, I'm stumped as to what prettier can do in this case. I'll try to dig through remark's changes to see if this is an explicitly intentional difference... in any case, this might be a case of closing a few bugs and introducing another one due to the dependency upgrade. I don't know whether that's still a net desirable outcome.

Reply:
So, it's about 

```markdown
`
 a
`
```

commonmark parse as `a`, and remark parse it as `<SPACE>a`, right?

Reply:
- [This PR](https://deploy-preview-8140--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEADAOlABAQ06kAGhAgAcYBLaAZ2VBwCcGIB3ABUYVpRwBsWcAT1rEARgxxgA1nBgBlUpIpQA5shgMArnGJwAtqLgATI8YAyOVZpwq4AMQgM9OGJVXIQOTTAhEQACxg9XgB1fwp4akUwODkuCIoANwjBDzBqERBlajgGGDYJFWdkADM+HOIAK2oADwAhCWlZORw9ODNlOFLynRBqmrllFV44AEVNCHhu3gqQRQYchg9nBikjVig-UgZlGBCKIxh-ZAAOAAZibYgckIlSD224RcSu4gBHCfgCsm5PagBaKBwYzGPwMOAfCjggo2YpIMozXo5PQUdRaJFDEbjSZdeE9YgwHCifaHY5IABMBIkFF4QwAwhA9HCQE8AKx+TQ5AAqRO4CNmiW0AEkoKZYHIwDtyABBUVyGCCEbTHIAXxVQA) 
- [Stable](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEADAOlABAQ06kAGhAgAcYBLaAZ2VBwCcGIB3ABUYVpRwBsWcAT1rEARgxxgA1nBgBlUpIpQA5shgMArnGIALGAFteAdV0V41RWDhyu5igDdzg5OGoiQy6nAYw2ElQMcZAAzPm9iACtqAA8AIQlpWTkcAzgAGWU4UPCdEGiYuWUVXjgARU0IeBzeCJBFBm8GVyCGKQATVigieoZlGGMKdphdZAAOAAZiUmZvYwlSVxm4Jods4gBHSvh-Mm4QHGoAWig4OHbznoY4LYpr-xxA4KQw2rzvAwp1LXfi0oqqtkXrliDAcKJBsNRkgAEygiQUXjFADCEAMQVcKwArD1NN4ACrg7ivOoObQASSgF1gcjAfXIAEEqXIYIJSjVvABfTlAA)

We have to check which one is correct then.

Reply:
Reported https://github.com/remarkjs/remark/issues/494